### ISSUE TITLE
Question about implementation of sinusoidal positional embedding

### ISSUE THREAD

**Issue by tae-mo** [lucidrains/denoising-diffusion-pytorch#76]
Is there any reason why not shuffling sin and cos values when calculating positional embedding?

Because to the best of my knowledge, "attention is all you need" used shuffled sinusoidal positional embedding values.

e.g.) sin, cos, sin, cos ...

But I see the implementation be like: torch.cat([sin,cos], dim=-1), which would be like: sin, sin, sin, ..., cos, cos, cos ...

I would appreciate if you let me know. Thanks!

**Comment by whubaichuan** [comment]
@tae-mo yes, the same problem as [here](https://www.youtube.com/watch?v=a4Yfz2FxXiY&lc=Ugz1q774TwKIYZ8D93J4AaABAg.9gaWftBtn1g9go4hP2N-Jr) I proposed. Is any solution?

**Comment by whubaichuan** [comment]
@tae-mo 

I replace the last two line with:

            output=torch.empty(embeddings.shape[0], embeddings.shape[1]*2) 
            for j in range(embeddings.shape[0]):
                for i in range(half_dim):
                        output[j,2*(i)]=embeddings.sin()[j,i]
                        output[j,2*i+1]=embeddings.cos()[j,i]
            return output
